submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,227 @@
+
+  To achieve this, one can apply the following CRD.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalnetEgressIP
+   metadata:
+     name: ns-egressip
+     namespace: myproject
+   spec:
+     globalIP: (optional)
+```
+
+3.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIP as egressIP for cross-cluster communication.
+
+  To achieve this, one can apply the following CRD.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalnetEgressIP
+   metadata:
+     name: db-pods
+     namespace: myproject
+   spec:
+     globalIP: (optional)
+     selector:
+       podSelector:
+         matchLabels:
+           role: db
+       priority: 100 (optional) 
+```
+
+4.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIP both as ingress and egressIP for cross-cluster communication.
+
+  To achieve this, user can create a Headless Service (backed by a Deployment or StatefulSets) and
+  Globalnet will ensure that all the backend Pods that belong to the Headless Service are assigned
+  a globalIP. User is also expected **not** to create any GlobalnetEgressIP object which selects the
+  backendPods that match the Headless Service.
+
+![GlobalnetTopology](./images/globalnet-headless-svc.png)
+
+  In the above example, each of the http pods will get its own globalIP which can be used for
+  both ingress and egress communication.
+
+  However, if the user wants egress traffic from the http Pods to carry a unique globalIP,
+  its possible by explicitly requesting a GlobalnetEgressIP with PodSelectors matching Headless
+  Service Pods as shown below.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalnetEgressIP
+   metadata:
+     name: http-pods
+     namespace: myproject
+   spec:
+     globalIP: (optional)
+     selector:
+       podSelector:
+         matchLabels:
+           app: http
+   status:
+     globalIP: 169.254.0.100
+```
+
+![GlobalnetTopology](./images/globalnet-svc.png)
+
+## Alternatives
+
+It is possible to support Pod to Pod communication even with Globalnet. However, to support this, it
+requires programming additional IPtable rules on the nodes which will have an impact on the performance
+of the solution. This proposal provides some limited Pod to Pod connectivity via the Headless Services.
+If user is interested in Pod to Pod communication for all the Pods, it is recommended to install
+Clusters with non-overlapping CIDRs and use Vanilla Submariner instead of Globalnet.
+
+## Work items
+
+1. Support annotation at the cluster level and program necessary egress rules.
+2. Support annotation at namespace level which takes precedence over globalIp at cluster level.
+3. Support annotation to Pods (or set of Pods) which has the highest precedence while
+   honouring the priority.
+4. Annotate only exported Services instead of annotating all Services.
```

Comment:
As mentioned earlier, since this is already done, can consider removing it.

Reply:
Lets not remove it, we are not backporting this feature to current GN release.